src/peft/peft_model.py

```diff
@@ -199,6 +203,25 @@ def save_pretrained(
             os.makedirs(output_dir, exist_ok=True)
 
             if safe_serialization:
+                # Safetensors does not allow tensor aliasing.
+                # We're going to remove aliases before saving
+                ptrs = collections.defaultdict(list)
+                for name, tensor in output_state_dict.items():
+                    # Sometimes in the state_dict we have non-tensor objects.
+                    # e.g. in bitsandbytes we have some `str` objects in the state_dict
+                    if isinstance(tensor, torch.Tensor):
+                        ptrs[id_tensor_storage(tensor)].append(name)
+                    else:
+                        # In the non-tensor case, fall back to the pointer of the object itself
+                        ptrs[id(tensor)].append(name)
```

Comment:
Not sure if this is needed, I added that in the transformers code to handle bnb state dict case and should be safe to keep it but happy to remove it as well

Reply:
Probably it's better to follow the same steps as transformers.